### ISSUE TITLE
Panic when e.g. IGNORE("@") or IGNORE("*") being used

### ISSUE THREAD

**Issue by patschi** [StackExchange/dnscontrol#799]
Just when testing around I've seen dnscontrol panic'ing when using invalid/specific IGNORE values, like:
```
IGNORE("@")
IGNORE("*")
```

Stacktrace of the panic:
```
panic: Trying to update/add IGNOREd record: @ MX

goroutine 1 [running]:
github.com/StackExchange/dnscontrol/v3/pkg/diff.(*differ).IncrementalDiff(0xc00081aa80, 0xc000228100, 0x1b, 0x20, 0x1e3b2c8, 0x0, 0x0, 0x1e3b2c8, 0x0, 0x0, ...)
        /go/src/github.com/StackExchange/dnscontrol/pkg/diff/diff.go:105 +0x257e
github.com/StackExchange/dnscontrol/v3/providers/ovh.(*ovhProvider).GetDomainCorrections(0xc00070e050, 0xc000459b80, 0x11, 0xc000459b00, 0x1, 0x0, 0x0)
        /go/src/github.com/StackExchange/dnscontrol/providers/ovh/ovhProvider.go:117 +0x1b0
github.com/StackExchange/dnscontrol/v3/commands.run(0x12dc303, 0xc, 0x0, 0x0, 0x0, 0x0, 0x0, 0x12da052, 0xa, 0x0, ...)
        /go/src/github.com/StackExchange/dnscontrol/commands/previewPush.go:135 +0x3cc
github.com/StackExchange/dnscontrol/v3/commands.Preview(...)
        /go/src/github.com/StackExchange/dnscontrol/commands/previewPush.go:87
github.com/StackExchange/dnscontrol/v3/commands.glob..func6.1(0xc0000a6740, 0xb, 0xc)
        /go/src/github.com/StackExchange/dnscontrol/commands/previewPush.go:25 +0x96
github.com/urfave/cli/v2.(*Command).Run(0xc0001217a0, 0xc0000a6400, 0x0, 0x0)
        /go/src/github.com/StackExchange/dnscontrol/vendor/github.com/urfave/cli/v2/command.go:164 +0x4e0
github.com/urfave/cli/v2.(*App).RunContext(0xc000082300, 0x15e40a0, 0xc0000320c0, 0xc00000c080, 0x2, 0x2, 0x0, 0x0)
        /go/src/github.com/StackExchange/dnscontrol/vendor/github.com/urfave/cli/v2/app.go:306 +0x814
github.com/urfave/cli/v2.(*App).Run(...)
        /go/src/github.com/StackExchange/dnscontrol/vendor/github.com/urfave/cli/v2/app.go:215
github.com/StackExchange/dnscontrol/v3/commands.Run(0xc0000ce060, 0x5e, 0xc00007c058)
        /go/src/github.com/StackExchange/dnscontrol/commands/commands.go:59 +0x23e
main.main()
        /go/src/github.com/StackExchange/dnscontrol/main.go:18 +0x40
```

While it's quite obvious when dnscontrol panic's and the fix is simply not using it, I don't assume it isn't intended that `IGNORE()` causes a panic when invalid value is provided?

**Comment by tlimoncelli** [comment]
Yeah, that shouldn't panic.  

`IGNORE` and `NO_PURGE` need a lot more testing.  They're hacky solutions to the problem of 2 systems trying to update a domain; which is a risky thing to do in the first place.

**Comment by patschi** [comment]
I believe this is now kinda-obsolete as #806 deprecates `IGNORE()` anyway, when I've understood that right, @tlimoncelli?

**Comment by mikenz** [comment]
@patschi #806 replaces it with the more specifically named `IGNORE_NAME()` but retains identical functionality (and probably bugs)

**Comment by patschi** [comment]
Just to ensure I'm not understanding it, but this issue is fixed by @mikenz in #817, isn't it?
